weighing.py

```diff
@@ -275,6 +289,8 @@ def analysis(cls, weighings):
             with_rec_name=False)
         product = Product(**default_product_values)
         for weighing in weighings:
+            if weighing.not_assigned_weight and not weighing.forced_analysis:
+                raise UserError(gettext('agronomics.msg_not_assigned_weight'))
```

Comment:
Always show the rec_name or some reference to the weighing that is wrong. Otherwise transitioning several weighings will result in a generic error and the user will not know which one is wrong.